[BUGFIX] execSync compat issue #92

joebartels opened the pull request:
branch is intended to resolve issues around `child_process` `execSync` compatibility among the nodejs versions and platforms it's installed on. [sync-exec](https://www.npmjs.com/package/sync-exec) is a good candidate. Another possible sol'n is latest version of [execSync](https://github.com/mgutz/execSync) as noted by @duizendnegen in https://github.com/ember-cli/ember-cli-deploy/issues/92

This PR uses [sync-exec](https://www.npmjs.com/package/sync-exec) and was confirmed to resolve https://github.com/ember-cli/ember-cli-deploy/issues/92 and is a potential fix for https://github.com/ember-cli/ember-cli-deploy/issues/90

lukemelia commented:
@joebartels Thanks for working on this. In the test, it looks like what you are stubbing doesn't match the usage. I think this is the source of the failure? Can you fix?

joebartels commented:
@lukemelia thanks for the feedback. I've updated how the sha test builds a stub for `sync-exec` and refactored `createTag` to work with how I changed the test... not proud of doing it like this. The other alternative I considered was instead of hardcoding a sha1 value in the sha test, it could be created from the actual `git rev-parse HEAD` at test runtime. e.g.,
```javascript
// sha-test.js
var syncExec = require('sync-exec');
...
var GIT_SHA = syncExec('git rev-parse HEAD').stdout;
```
then no sandboxing is required for `sync-exec`. Although since it wasn't done this way I decided against it.  Open to any suggestions (or PRs) :)

joebartels commented:
or create a utility, `utilities/sync-exec.js` that exports an object with the function `require('sync-exec')` as a method, that way it can be stubbed nicely: https://github.com/joebartels/ember-cli-deploy/commit/fceeca3a0f21b3d9396d92be9bfa55fe0cba43b0

lukemelia commented:
Left you a PR @joebartels

lukemelia commented:
@joebartels After you merge my PR, go ahead and squash it down to one commit with a descriptive message, and I'll merge it in the morning.

joebartels commented:
:+1: really appreciate the help @lukemelia
